lib/internal/fs/streams.js

```diff
@@ -100,46 +101,17 @@ function ReadStream(path, options) {
     }
   }
 
-  if (typeof this.fd !== 'number')
-    this.open();
-
-  this.on('end', function() {
-    if (this.autoClose) {
-      this.destroy();
-    }
+  this.on('ready', function() {
+    // Start the flow of data.
+    this.read();
   });
+
+  Readable.call(this, options);
```

Comment:
This needs to be called after all values needed from `_construct` are initialized but before any Readable methods are needed.